commit d2e2d00094019028e553bfdd7f0bb18eefd1d5fe
Author: Mike Frysinger <vapier@gmail.com>
Date:   Sat Aug 8 01:33:28 2015 -0400

    drop use of datestamp in compiled output
    
    This doesn't add that much (if any) value to the version output, and it
    prevents systems from caching the results or reproducing it later on.
    Drop it all.
    
    Fixes #3.

Index: ncompress-4.2.4.4/Makefile.def
===================================================================
--- ncompress-4.2.4.4.orig/Makefile.def
+++ ncompress-4.2.4.4/Makefile.def
@@ -38,7 +38,7 @@ LBOPT= $(LDFLAGS)
 
 
 compress:	Makefile compress42.c patchlevel.h
-	$(CC) -o compress $(options) "-DCOMPILE_DATE=\"`date`\"" compress42.c $(LBOPT)
+	$(CC) -o compress $(options) compress42.c $(LBOPT)
 
 install:	compress
 		[ -f $(DESTDIR)$(BINDIR)/compress ] && \
Index: ncompress-4.2.4.4/build
===================================================================
--- ncompress-4.2.4.4.orig/build
+++ ncompress-4.2.4.4/build
@@ -566,9 +566,9 @@ do
 		echo ""
 		echo "Compiling compress ${version}"
 
-		echo ${CC} ${options} "-DCOMPILE_DATE=\"`date`\"" compress42.c ${LBOPT}
+		echo ${CC} ${options} compress42.c ${LBOPT}
 
-		if ${CC} ${options} "-DCOMPILE_DATE=\"`date`\"" compress42.c ${LBOPT}
+		if ${CC} ${options} compress42.c ${LBOPT}
 		then
 			echo ""
 			./compress -V
@@ -599,9 +599,9 @@ do
 		echo ""
 		echo "Compiling compress ${version}"
 
-		echo ${CC} ${options} "-DCOMPILE_DATE=\"`date`\"" compress42.c 
+		echo ${CC} ${options} compress42.c 
 
-		${CC} ${options} "-DCOMPILE_DATE=\"`date`\"" compress42.c 
+		${CC} ${options} compress42.c 
 		echo ""
 		echo ${n} "Press return to continue${c}"
 		read dummy
@@ -790,7 +790,7 @@ LBOPT=${LBOPT}
 
 
 compress:	Makefile compress42.c patchlevel.h
-	$(CC) -o compress $(options) "-DCOMPILE_DATE=\"\`date\`\"" compress42.c $(LBOPT)
+	$(CC) -o compress $(options) compress42.c $(LBOPT)
 
 install:	compress
 		[ -f \$(BINDIR)/compress ] && \\
Index: ncompress-4.2.4.4/compress42.c
===================================================================
--- ncompress-4.2.4.4.orig/compress42.c
+++ ncompress-4.2.4.4/compress42.c
@@ -261,7 +261,6 @@
 #	define	BYTEORDER 	4321
 #	undef	NOALLIGN
 #	define	NOALLIGN	1
-#	define	COMPILE_DATE	__DATE__
 #endif /* DOS */
 
 #ifndef	O_BINARY
@@ -1865,7 +1864,7 @@ prratio(stream, num, den)
 void
 about()
 	{
-		fprintf(stderr, "Compress version: %s, compiled: %s\n", version_id, COMPILE_DATE);
+		fprintf(stderr, "Compress version: %s\n", version_id);
 		fprintf(stderr, "Compile options:\n        ");
 #if BYTEORDER == 4321 && NOALLIGN == 1
 		fprintf(stderr, "USE_BYTEORDER, ");
